packages/smooth_app/lib/pages/prices/infinite_scroll_controller.dart

```diff
@@ -0,0 +1,104 @@
+import 'dart:async';
+
+/// A generic controller for handling infinite scrolling in lists.
+/// [T] is the type of items being displayed.
+/// [P] is the type of parameters used for pagination.
+class InfiniteScrollController<T, P> {
+  InfiniteScrollController({
+    required this.fetchItems,
+    Iterable<T> initialItems = const <Never>[],
```

Comment:
Not very familiar with `Never`, but wouldn't that be similar to the following code?
```suggestion
    required Iterable<T> initialItems,
```